pd/src/components/staking.rs

```diff
@@ -361,14 +361,26 @@ impl Component for Staking {
             base_exchange_rate: 1_0000_0000,
         };
         self.overlay
-            .set_base_rates(cur_base_rate, next_base_rate)
+            .set_base_rates(cur_base_rate.clone(), next_base_rate)
             .await;
 
+        let mut allocations_by_validator = HashMap::new();
+        for allocation in &app_state.allocations {
+            if allocation.amount == 0 {
+                continue;
+            }
+
```

Comment:
non-blocking: this could be cut out, since we'll just be adding 0 to the total, so it's a no-op